xonsh/completers/path.py

```diff
@@ -151,6 +151,59 @@ def _quote_paths(paths, start, end):
     return out
 
 
+def joinpath(path):
+    if path is None:
+        return ''
+    if len(path) == 0:
```

Comment:
I think in the bytecode, they are likely to be identical in this case, because `return` cuts of all execution (we can only get down to the second `if` if the first condition was false).